ncm-ceph/src/main/perl/Ceph/ClusterMap.pm

```diff
@@ -0,0 +1,229 @@
+#${PMpre} NCM::Component::Ceph::ClusterMap${PMpost}
+
+use parent qw(CAF::Object);
+use Readonly;
+use Data::Dumper;
+use JSON::XS;
+
+
+sub _initialize
+{
+    my ($self, $clusterobj) = @_;
+
+    $self->{log} = $clusterobj;
+
+    $self->{quattor} = {};
+    $self->{ceph} = {};
+
+    $self->{Cluster} = $clusterobj;
+    $self->{deploy} = {};
+    return 1;
+}
+
+sub get_fqdn
+{
+    my ($self, $host) = @_;
+    return $self->{quattor}->{$host}->{fqdn};
+};
+
+# get the parsed daemons as they are declared in schema
+sub get_quattor_map
+{
+    my ($self) = @_;
+    return $self->{quattor};
+}
+
+# Gets the MON map
+sub mon_hash
+{
+    my ($self) = @_;
+    my ($ec, $jstr) = $self->{Cluster}->run_ceph_command([qw(mon dump)], 'get mon map', nostderr => 1) or return;
+    my $monsh = decode_json($jstr);
+    foreach my $mon (@{$monsh->{mons}}){
+        $self->add_existing('mon', $mon->{name}, { addr => $mon->{addr}});
+    }
+    return 1;
+}
+
+# Gets the MGR map
+sub mgr_hash
+{
+    my ($self) = @_;
+    my ($ec, $jstr) = $self->{Cluster}->run_ceph_command([qw(mgr dump)], 'get mgr map', nostderr => 1) or return;
+    my $mgrsh = decode_json($jstr);
+    $self->add_existing('mgr', $mgrsh->{active_name});
+    foreach my $mgr (@{$mgrsh->{standbys}}){
+        $self->add_existing('mgr', $mgr->{name});
+    }
+    return 1;
+}
+
+# Gets the MDS map
+sub mds_hash
+{
+    my ($self) = @_;
+    my ($ec, $jstr) = $self->{Cluster}->run_ceph_command([qw(mds stat)], 'get mds map', nostderr => 1) or return;
+    my $mdshs = decode_json($jstr);
+    my $fsmap = $mdshs->{fsmap};
+    foreach my $fs (@{$fsmap->{filesystems}}){
+        foreach my $mds (values %{$fs->{mdsmap}->{info}}) {
```

Comment:
loop over sorted keys

Reply:
why keys? I dont need them

Reply:
you can loop over sorted values if you want, as long as it reproducable. if you don;t need the keys, use a list?

Reply:
this is json output of ceph, keys are pids or someting. I'll sort the values then